pallets/payment/src/tests.rs

```diff
@@ -27,7 +27,8 @@ fn test_create_payment_works() {
 				amount: 20,
 				incentive_amount: 2,
 				state: PaymentState::Created,
-				resolver_account: RESOLVER_ACCOUNT
+				resolver_account: RESOLVER_ACCOUNT,
+				fee_detail: (FEE_RECIPIENT_ACCOUNT, 0)
```

Comment:
Perhaps it's worth having a test case for an implementation of the fee handler that actually returns a fee so we can catch issues like the unreserve above ;)

Reply:
The `test_charging_fee_payment_works` already exists but it was asserting wrong results 😅 I've fixed that and also added `test_charging_fee_payment_works_when_canceled`